### PR TITLE
Revamp plugin documentation & strip plugin publishing functionality

### PR DESCRIPTION
**Pull request by lucaswoj**
This PR closes https://github.com/mapbox/mapbox-gl-js/issues/3947, closes https://github.com/mapbox/mapbox-gl-js/issues/3687, and removes 74,231 lines from source control 😄 

Definitely looking for feedback on the design of the plugins directory & taxonomy of plugins. 

Any design suggestions @tristen @mayagao? 

screen shot 2017-02-20 at 3 57 55 pm

Related to https://github.com/mapbox/mapbox-gl-directions/pull/121

### CODE REVIEW

File: docs/_posts/examples/3400-01-16-point-from-geocoder-result.html [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 category: example
 title: Set a point after Geocoder result
-description: 'Listen to the <code>geocoder.input</code> event from the <a target="_blank" href="https://www.mapbox.com/mapbox-gl-js/plugins/#mapbox-gl-geocoder">Geocoder plugin</a> and place a point on the coordinate results.'
+description: 'Listen to the <code>geocoder.input</code> event from the <a target="_blank" href="https://github.com/mapbox/mapbox-gl-geocoder">Geocoder plugin</a> and place a point on the coordinate results.'
 tags:
   - controls-and-overlays
 ---
-<script src='https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-geocoder/v{{site.data.plugins.mapbox-gl-geocoder.latest}}/mapbox-gl-geocoder.js'></script>
-<link rel='stylesheet' href='https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-geocoder/v{{site.data.plugins.mapbox-gl-geocoder.latest}}/mapbox-gl-geocoder.css' type='text/css' />
+<script src='https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-geocoder/v2.0.1/mapbox-gl-geocoder.js'></script>
```
Comment (mollymerp) [inline docs/_posts/examples/3400-01-16-point-from-geocoder-result.html]: What's the plan for keeping these versions up to date?

Comment (lucaswoj) [inline docs/_posts/examples/3400-01-16-point-from-geocoder-result.html]: The plan is for these to be manually updated alongside plugin releases, as they are now.

**Comment by lucaswoj** [comment]
> Have you added publishing scripts to CI for all the Mapbox supported plugins?

Not yet. Waiting for final 👍 on the `mapbox-gl-directions` PR before copying that code across all of our plugins.

**Comment by tristen** [comment]
LGTM! Tiny nit but I'd make a couple spacing changes to define these groups better:

screen shot 2017-02-21 at 10 34 18 am

**Comment by lucaswoj** [comment]
# Todo Before Merging 

 - [ ] merge https://github.com/mapbox/mapbox-gl-directions/pull/121
 - [ ] add s3 deployment script to `mapbox-gl-compare`
 - [ ] add s3 deployment script to `mapbox-gl-draw`
 - [ ] add s3 deployment script to `mapbox-gl-geocoder`
 - [ ] add s3 deployment script to `mapbox-gl-rtl-text`
 - [ ] add s3 deployment script to `mapbox-gl-supported`
 - [ ] implement design tweaks https://github.com/mapbox/mapbox-gl-js/pull/4305#issuecomment-281379197
 - [ ] pull in additional changes from #4308 #4310

**Comment by mcwhittemore** [comment]
Maybe as a next step, it would be cool if the script that sends the distribution file to s3, also created a version file that could be used for auto updating the version numbers here. That said, this is a great leap forward towards making releases better. Thanks!

**Comment by lucaswoj** [comment]
@mollymerp I finished adding deployment processes to all existing Mapbox plugins. Can you rereview? :pray:

**Comment by mollymerp** [comment]
Nice work @lucaswoj! 💪 ✨

**Comment by andrewharvey** [comment]
@lucaswoj With this merged, does it make sense to remove the list in the README.md and simply point to https://www.mapbox.com/mapbox-gl-js/plugins/ ? For example the mapbox-gl-directions plugin is missing in README.md so the risk of them becoming out of sync is real.

**Comment by lucaswoj** [comment]
Yes @andrewharvey. That was my intention. Submitting a PR for the `README.md` in a moment.

**Pull request merged**